Add "Deploy to Heroku" button

creatorrr opened the pull request.

creatorrr committed:
Added Deploy to Heroku button

Signed-off-by: Diwank Singh <diwank.singh@gmail.com>

creatorrr committed:
Fix app.json validation error

Signed-off-by: Diwank Singh <diwank.singh@gmail.com>

creatorrr committed:
Make AWS_S3_HOST optional

Signed-off-by: Diwank Singh <diwank.singh@gmail.com>

creatorrr committed:
Move to python2

Signed-off-by: Diwank Singh <diwank.singh@gmail.com>

creatorrr committed:
Fixed env vars

Signed-off-by: Diwank Singh <diwank.singh@gmail.com>

creatorrr committed:
Add heroku URL to ALLOWED_HOSTS

Signed-off-by: Diwank Singh <diwank.singh@gmail.com>

creatorrr committed:
Added domain whitelist to app.json

Signed-off-by: Diwank Singh <diwank.singh@gmail.com>

talentoscope reviewed (changes requested):
The changes requiring a hobby account with Heroku should not be merged as that account level is not required.
Also, it is worth noting that ffmpeg is a requirement of transcroobie too, so this deployment will likely not work in its current state.
File upload is also not possible with Heroku, so the GOOGLE_APP_CONFIG_DATA and GOOGLE_APPLICATION_CREDENTIALS are only able to be uploaded by exposing these publicly unless one has a paid account with github.
See my fork for changes made to base64 encoding the json file into an envvar to work around this limitation.

creatorrr committed:
Update app.json

creatorrr committed:
Added ffmpeg buildpack

creatorrr commented:
Replied inline, @xteejx

> The changes requiring a hobby account with Heroku should not be merged as that account level is not required.

Agreed. fixed.

> Also, it is worth noting that ffmpeg is a requirement of transcroobie too, so this deployment will likely not work in its current state.

Changed the buildpack to one that includes ffmpeg.

> See my fork for changes made to base64 encoding the json file into an envvar to work around this limitation.

Yeah I was wondering about that. Your fork seems to have significant changes though. Should I merge them?

talentoscope commented:
Awesome.
I think it's probably best to just cherry pick from my branch, as you say there are quite a few changes.
The ones in particular that I think would be useful to merge are the changes from hardcoded variables to envvars. Perhaps the qualification requirement and submit timer too, as in production testing, I found this avoids abuse by Turkers just hitting submit on the first CHECK round, thereby causing real headaches, and the qualification avoids inexperienced Turkers from jumping on the HITs.

What would be perfect would be some quality control tracking using qualifications, which I've discussed with Armin (the author).
